src/lib/sandbox/main-forward-trigger.ts

```diff
@@ -17,6 +21,19 @@ export const mainForwardTrigger = (worker: PartytownWebWorker, $winId$: WinId, w
         $args$: serializeForWorker($winId$, Array.from(args)),
       },
     ]);
+  }
+
+  // Read config and execute a splice instead of the push method on the given global object
+  const pushBackData = (forwardArr: string[], args: any) => {
+    if (pushBacks.length > 0 && pushBacks.includes(forwardArr.join('.'))) {
+      if (forwardArr.length === 2 && forwardArr[1] === 'push') {
```

Comment:
Are we able to make this more dynamic?
`forwardArr.length === 2 && forwardArr[1] === 'push'`

By hardcoding 'push' it limits the other 3rd party scripts that might also push to the window object.

This is a similar solution that I created:

```js function setPropertyOnWindow(propertyValue: any, windowProperties: any[], browserWindow: MainWindow, allArgs: any){
  let lastObjectPropertySet = browserWindow;

  windowProperties.forEach((property: any, index, array) => {
    if(windowProperties[0] === 'dataLayer' && windowProperties[1] === 'push' && index === array.length -1){
      browserWindow.dataLayer = browserWindow.dataLayer || [];
      browserWindow.dataLayer = [...browserWindow.dataLayer, propertyValue]
    } else if(index === array.length -1){
        lastObjectPropertySet[property] = lastObjectPropertySet[property] || propertyValue
    } 
    else {
      lastObjectPropertySet = lastObjectPropertySet[property] || {}
    }
  })
}```